inst/rmarkdown/templates/trundler_template/skeleton/skeleton.Rmd

```diff
@@ -3,74 +3,117 @@ title: "Trundler Report Template"
 ```
 
-## Import libraries
-
 ```{r libraries}
 library(trundler) 
 library(dplyr) 
 library(stringr)
 library(purrr)
 library(tidyr)
 library(ggplot2) 
+library(gt)
+library(glue)
 ```
 
-## Set the API key
-
-If you don't have an API key yet, you can get one from [RapidAPI](https://rapidapi.com/datawookie/api/trundler).
-
-```{r}
+This report demonstrates how you can quickly generate a report using the Trundler R package.    
+    
+```{r, include = FALSE}
+# You can get an API key from https://rapidapi.com/datawookie/api/trundler
 API_KEY <- Sys.getenv("TRUNDLER_KEY")
 set_api_key(API_KEY)
 ```
 
-## View catalog of retailers
-
-Here's a look at the first 10 retailers in the catalog
-
 ```{r}
 # Get a dataframe of all available retailers
-retailer_df <- retailer()
+retailer_df <- retailer() 
+```
+
+## View the retailer catalog
+
+Trundler collects data from wide range of retailers.  Currently, there are `r length(retailer_df$retailer)` retailers in the Trundler catalog.  Here are the first ten.
 
+<br>
+```{r}
 # View the first 10 retailers
-head(retailer_df, n = 10)
+gt(retailer_df[1:10, 1:4]) %>% 
+  tab_header(title = md("**Trundler Retailer Catalog**"),
+            subtitle = md("*The first ten retailers in the catalog*")) %>%
+  cols_label(retailer_id = md("**Retailer ID**"),
+             retailer = md("**Retailer**"),
+             retailer_url = md("**URL**"),
+             currency = md("**Currency**"))
 ```
 
-## Retrieve data for specific products
+<br>
+
+## Track the price of a product at a specific retailer
+
+It's possible to search a retailer for a product of interest.  Here, we've searched for Bovril products sold at Pick 'n Pay.
 
-Let's say you're interested in the price of Bovril at Pick 'n Pay.  You can use the retailer ID for Pick 'n Pay (9) to search for all Bovril products at Pick 'n Pay.
+<br>
 
 ```{r}
-# Search for Bovril products.
-retailer_products(retailer_id = 9, product = "Bovril")
+# Use the retailer_id to search for Bovril products at a specific retailer (e.g. Pick 'n Pay)
+all_bovril <- retailer_products(retailer_id = 9, product = "Bovril") %>%
+  select(product_id, product, sku)
+
+# Print table
+gt(all_bovril) %>%
+  tab_header(title = md("**All Bovril Products**"),
+             subtitle = md("*At Pick 'n Pay*")) %>%
+  cols_label(product_id = md("**Product ID**"),
+             product = md("**Product**"),
+             sku = md("**SKU**"))
+
 ```
+<br>
 
-Pick 'n Pay sells a few different Bovril products.  Let's choose 'Bovril Beefy Spread 250g'.  Using the corresponding product ID, you can get the product and price data you're interested in.
+As you can see, Pick 'n Pay sells a few different Bovril products.  Let's have a look at the price of the 250g product.  Here are the most recent price records.
 
+<br>
 ```{r}
-# Get the product data 
+# Using the corresponding product ID, you can get the product and price data you're interested in. 
 bovril_250g <- product(543374) 
```

Comment:
Rather than hard coding 543374 can you get this product ID directly from the data using filtering?